docs/dump_javadump.md

```diff
@@ -660,6 +678,206 @@ NULL           =================================
+2XMPOOLDAEMON      Current total number of live daemon threads: 15
+NULL
+1XMCURTHDINFO  Current thread
+3XMTHREADINFO      "main" J9VMThread:0xB6B60C00, omrthread_t:0xB6B049D8, java/lang/Thread:0x95764520, state:R, prio=5
+3XMJAVALTHREAD            (java/lang/Thread getId:0x1, isDaemon:false)
+3XMTHREADINFO1            (native thread ID:0x49C6, native priority:0x5, native policy:UNKNOWN, vmstate:R, vm thread flags:0x00001020)
+3XMTHREADINFO2            (native stack address range from:0xB6CB5000, to:0xB74B6000, size:0x801000)
+3XMCPUTIME               CPU usage total: 8.537823831 secs, current category="Application"
+3XMHEAPALLOC             Heap bytes allocated since last GC cycle=0 (0x0)
+3XMTHREADINFO3           Java callstack:
+4XESTACKTRACE                at java/lang/StringBuffer.ensureCapacityImpl(StringBuffer.java:696)
+4XESTACKTRACE                at java/lang/StringBuffer.append(StringBuffer.java:486(Compiled Code))
+5XESTACKTRACE                   (entered lock: java/lang/StringBuffer@0x957645B8, entry count: 1)
+4XESTACKTRACE                at java/lang/StringBuffer.append(StringBuffer.java:428(Compiled Code))
+4XESTACKTRACE                at HeapBreaker.main(HeapBreaker.java:34(Compiled Code))
+3XMTHREADINFO3           Native callstack:
+4XENATIVESTACK               (0xB6C535B3 [libj9prt29.so+0x3b5b3])
+4XENATIVESTACK               (0xB6C36F3E [libj9prt29.so+0x1ef3e])
+4XENATIVESTACK               (0xB6C5311E [libj9prt29.so+0x3b11e])
+4XENATIVESTACK               (0xB6C53216 [libj9prt29.so+0x3b216])
+4XENATIVESTACK               (0xB6C52CE3 [libj9prt29.so+0x3ace3])
+4XENATIVESTACK               (0xB6C36F3E [libj9prt29.so+0x1ef3e])
+4XENATIVESTACK               (0xB6C52C57 [libj9prt29.so+0x3ac57])
+4XENATIVESTACK               (0xB6C4E9CD [libj9prt29.so+0x369cd])
+4XENATIVESTACK               (0xB6C502FA [libj9prt29.so+0x382fa])
+```
+
+To simulate a Java `OutOfMemoryError`, our testcase works by repeatedly appending characters to a `StringBuffer` object in an infinite loop. The Java callstack shows the `HeapBreaker.main` method appending characters (`java/lang/StringGuffer.append`) until the method `java/lang/StringBuffer.ensureCapacityImpl()` throws the `OutOfMemoryError`.
+
+StringBuffer objects are wrappers for character arrays (`char[]`) and when the capacity of the underlying array is reached, the contents are automatically copied into a new, larger array. The new array is created in the `StringBuffer.ensureCapacity()` method, which more or less doubles the size of the old array. In our scenario, the array takes up all the remaining space in the Java heap.
+
+Although the Java code we used in this scenario was deliberately triggering an `OutOfMemoryError` in a spectacular way, similar allocation issues can and do occur when dealing with large data sets such as XML files.
+
+The next step in diagnosing the problems would be to open the system dump that gets generated automatically when an `OutOfMemoryError` occurs. Open
```

Comment:
Would the gc history section help in this case? Showing the amount of heap memory being allocated which lead to the OOM.

Reply:
Yes, that information is useful for determining whether the problem was caused by a full heap or an unexpectedly large allocation request.

Reply:
@pshipton - I've added a short paragraph about the GC History section. Please can you check it to make sure I have not misinterpreted the output.